experiments/react-native-uifabric/src/components/PersonaCoin/PersonaCoin.types.ts

```diff
@@ -0,0 +1,93 @@
+import { IViewWin32Props } from '@office-iss/react-native-win32';
+import { ImageProps, ViewProps, ImageURISource, TextProps } from 'react-native';
+import { IBackgroundColorTokens, IForegroundColorTokens } from 'src/tokens';
+import { IRenderData } from '@uifabricshared/foundation-composable';
+
+export const personaCoinName = 'RNFPersonaCoin';
+
+export enum PersonaSize {

```

Comment:
We should get rid of all enums in our interface files.  Instead they should be discrete unions so:

`export type PersonaSize = 'size4' | 'size8' | 'size24' | ...`

This has the advantage of actually being strings so elsewhere where you have switch statements they can instead become direct mappings.  I __think__ you can then do something like:

    interface IMyTable {
      [key: keyof PersonaSize]: { param1: string, param2: string }
    };

The type can be direct indexed using the string value and typescript will complain if you change the PersonaSize type.  (I think)  You can test it, if it doesn't then you might have to define it as a mapped type so:

   type IMyObject = { [K in keyof PersonaSize]: { param1: string, param2: string } };

Either way we shouldn't use TS enums, they are pure overhead and the fact that they become numbers means that mismatched versions will cause mismatched indexing into enums if a new value is added.

Reply:
I can change it, but are we OK with diverging from the web Persona component, which uses enum?